js/components/modal.html

```diff
@@ -16,6 +17,14 @@
 					}
 				"></div>
 			</div>
+			<div class="modal-footer" data-bind="visible: $component.footerTemplate">
+				<div data-bind="
+					template: {
+						name: $component.footerTemplate,
+						data: $component.footerData
```

Comment:
Why do you need separate prop for `footerData`? I would use single `data` object